### PR TITLE
Introduce V28 consensus version

### PR DESCRIPTION
**Pull request by algorandskiy**
* TEAL v4
* Larger programs
* Larger app/asset lookup limits
* Longer asset URL
* Fee pooling within a group
* Keyreg txn additional checks

Removed InitialRewardsRateCalculation and PaysetCommit from vFuture since it is already in v26

Fixed some tests after exposing zero fees and strict keyreg as current consensus

### CODE REVIEW

File: config/consensus.go [hunk 1]
```diff
@@ -940,25 +960,9 @@ func initConsensusProtocols() {
...
 	vFuture.CompactCertSecKQ = 128
 
-	vFuture.EnableKeyregCoherencyCheck = true
-
-	// Enable support for larger app program size
-	vFuture.MaxExtraAppProgramPages = 3
-	vFuture.MaxAppProgramLen = 2048
-
-	// enable the InitialRewardsRateCalculation fix
-	vFuture.InitialRewardsRateCalculation = true
```
Comment (tsachiherman) [inline config/consensus.go:950]: the following `vFuture.PaysetCommit = PaysetCommitMerkle` can also be dropped.

**Comment by onetechnical** [comment]
For #2263 do we also need:

```
	vFuture.MaxAppTxnForeignApps = 8
	vFuture.MaxAppTxnForeignAssets = 8
```

File: config/consensus.go (hunk 1 above)
Comment (algorandskiy) [reply config/consensus.go:950]: done

**Review by tsachiherman — APPROVED** [review]
looks good to me.

**Review by jannotti — APPROVED** [review]
I don't know the part keys stuff, but the included things in v28 looks right to me.

**Pull request merged**